cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -1269,11 +1301,23 @@ var _ = Describe("Operation Controller", func() {
 					appSync: func(ctx context.Context, s1, s2, s3 string, c client.Client, cs *utils.CredentialService, b bool) error {
 						return nil
 					},
+					refreshApp: refreshApplication,
 				}
 
 				retry, err := task.PerformTask(ctx)
 				Expect(err).Should(BeNil())
 				Expect(retry).To(BeFalse())
+
+				By("verify if the refresh annotation is removed")
+				Eventually(func() bool {
+					err = k8sClient.Get(ctx, client.ObjectKeyFromObject(applicationCR), applicationCR)
+					if err != nil {
+						GinkgoWriter.Println(err)
+						return false
+					}
+					return applicationCR.Annotations == nil
+				}, "10s", "1s").Should(BeTrue())
+
```

Comment:
I think we can also add a check to make sure that the refresh annotation was added by the call to `PerformTask`: for example, have `refreshHandler` set a bool when it detects that refresh annotation was added, then Expect on that bool here.